$OpenBSD: patch-src_winwidget_c,v 1.11 2012/12/05 13:44:35 dcoppa Exp $

Do not apply --scale-down to a thumbnail window
(upstream git commit 7c00adf853acc76520d43a12eb63ec151924d5ab)

--- src/winwidget.c.orig	Tue Oct 16 06:31:41 2012
+++ src/winwidget.c	Wed Dec  5 15:26:34 2012
@@ -402,6 +402,7 @@ void winwidget_render_image(winwidget winwid, int resi
 
 	if (!winwid->full_screen && opt.scale_down && ((winwid->w < winwid->im_w)
 						       || (winwid->h < winwid->im_h)) &&
+							  (winwid->type != WIN_TYPE_THUMBNAIL) &&
 							  (winwid->old_zoom == 1.0)) {
 		D(("scaling down image %dx%d\n", winwid->w, winwid->h));
 
@@ -512,7 +513,8 @@ void winwidget_render_image(winwidget winwid, int resi
 			winwid->im_y = (int) (max_h - (winwid->im_h * winwid->zoom)) >> 1;
 		}
 	}
-	else if (need_center && !winwid->full_screen && opt.scale_down) {
+	else if (need_center && !winwid->full_screen && opt.scale_down
+			&& (winwid->type != WIN_TYPE_THUMBNAIL)) {
 		winwid->im_x = (int) (winwid->w - (winwid->im_w * winwid->zoom)) >> 1;
 		winwid->im_y = (int) (winwid->h - (winwid->im_h * winwid->zoom)) >> 1;
 	}
